packages/playback/src/lib/types/configuration.declarations.ts

```diff
@@ -61,10 +62,120 @@ export interface PlayerHlsConfiguration {
   customTagMap: CustomTagMap;
   transformTagValue: TransformTagValue;
   transformTagAttributes: TransformTagAttributes;
+
+}
+
+/** Would be used if we allowed the user to pass in a persistent session */
+// If we need this, move it to another file
+// export interface KeySessionMetadata {
+//   sessionId: string;
+//   initData: Uint8Array;
+//   initDataType: string;
+// }
+
+
+export interface PlayerEmeConfiguration {
+  /**
+   * A map of ClearKey key IDs to keys.
+   * These values should be encoded in hex or base64.
+   * Defaults to an empty object.
+   */
+  clearKeys: Record<string, string>;
+
+  /**
+   * Key sessions metadata to load before starting playback.
+   * Defaults to an empty array.
+   */
+  // initialKeySessions: Array<KeySessionMetadata>;
+
+
+  // NOTE: We could make this order the priority that the user intends.
+  // If not, do we want a different config option that lists priority of key systems?
+  // Or maybe a `priority` value in the keySystems?
+  // Do we want to priortize different types of the same DRM (com.widevine.something vs com.widevine.alpha)
+  keySystems: Record<string, IKeySystemConfig>;
+
+
+  /**
+   * The time in ms to check if the media key session is expired.
+   * Defaults to 1000.
+   */
+  sessionExpirationInterval: number;
+
+  /**
+   * The minimum version of HDCP to start EME streams.
+   * Defaults to ''.
+   */
+  minHdcpVersion: string;
+
+  // This will probably replace the firstWebkitneedkeyTimeout option in contrib eme
+  /**
+   * Option to ignore duplicate init data on the encrypted event or through pssh boxes.
+   * Defaults to true
+   */
+  ignoreDuplicateInitData: boolean;
+
+
+  /**
+   * The amount of time in milliseconds to wait on the first `webkitneedkey` event before making the key request.
+   * is was implemented due to a bug in Safari where rendition switches at the start of playback can cause `webkitneedkey` to fire multiple times, with only the last one being valid.
+   * Defaults to 1000.
+   */
+  webkitneedkeyTimeout: number;
+
+  // POLYFILLS
+
+  // This would enable the legacy fairplay specific code, in a polyfill manner.
+  /**
+   * Build modern EME into browsers that use Apple's prefixed EME in Safari.
+   * Defaults to false
+   */
+  enableLegacyFairplay: boolean;
```

Comment:
not sure we need it, we should always polyfill older API

Reply:
Gotcha, good to know we should add pollyfills automatically.

Reply:
not all, but this specific - yes :)

Reply:
Consider it as "legacy-to-spec-adapter" instead of polyfill :)